sidecar/register/registration.go

```diff
@@ -82,12 +83,14 @@ func (agent *RegistrationAgent) Stop() {
 
 func (agent *RegistrationAgent) register() {
 	for {
-		registeredInstance, err := agent.config.Client.Register(agent.config.ServiceInstance)
-		if err == nil {
-			go agent.renew(registeredInstance)
-			return
+		if agent.checkAppHealth(DefaultReregistrationDelay) {
```

Comment:
Ah, never mind. By starting and stopping the registration agent from the health checker, you can ensure that the health state information is transmitted as soon as it is detected. That is a vast improvement.